pkg/proc/gdbserial/rr.go

```diff
@@ -20,7 +20,7 @@ import (
 // program. Returns a run function which will actually record the program, a
 // stop function which will prematurely terminate the recording of the
 // program.
-func RecordAsync(cmd []string, wd string, quiet bool, redirects [3]string) (run func() (string, error), stop func() error, err error) {
+func RecordAsync(cmd []string, wd string, quiet bool, redirects [3]proc.OutputRedirect) (run func() (string, error), stop func() error, err error) {
```

Comment:
It's weird that the stdin redirect is of type OutputRedirect, and that it can never be anything but a path. I'd say this should be changed to `stdinRedirect string, stdoutRedirect, stderrRedirect proc.OutputRedirect`. The stdin redirect isn't treated like the other two at all, at this point.

Reply:
Maybe the struct name should be changed back to `Redirect`.I want to keep `stdin` passed in as `*os.File`.

Reply:
Why? Nothing is using the *os.File version of stdin. Looking at the DAP specification I don't see anything that could use it, but maybe I'm missing it.

Reply:
I went through the `Dap` specification and didn't find definition of remote input event. Maybe there will be a specification to support remote input events.